timeago_flutter/lib/timeago_flutter.dart

```diff
@@ -64,14 +64,14 @@ class Timeago extends TimerRefreshWidget {
 
   final TimeagoBuilder builder;
   final DateTime date;
-  final DateTime clock;
+  final DateTime Function() clock;
   final String locale;
   final bool allowFromNow;
 
   @override
   Widget build(BuildContext context) {
     final formatted = timeago.format(date,
-        locale: locale, clock: clock, allowFromNow: allowFromNow);
+        locale: locale, clock: clock?.call(), allowFromNow: allowFromNow);
     return builder(context, formatted);
```

Comment:
This Introduce a breaking change. And that breaking change must come with null-safety.

Consider to add a new input argument  
  
\`\`\`dart  
final DateTime clock;  
final DateTime Function(BuildContext context, DateTime clock) clockBuilder;  
\`\`\`

and in the build method:  
  
\`\`\`dart  
final formatted = timeago(date, locale: locale, clock: clockBuilder?.call(context, clock) ?? clock,  
       allowFromNow: allowFromNow);  
\`\`\`\`  
  
But first, please provide a use case where this worth the change.